app/Http/Controllers/AdminDash.php

```diff
@@ -1357,6 +1357,11 @@ public function newEvent() {
         return view('dashboard.admin.events.new');
     }
 
+    public function scanForEvents() {
+        Artisan::call('Events:UpdateSupportEvents', []); // start a background task to update support events
+        return redirect('/dashboard/controllers/events/')->with('success', 'Support events updated successfully.');
```

Comment:
I would recommend changing this message to `Support events update began successfully. Updates may take up to an hour to process` (adjust time here to longer than actual). Just so we don't end up with something getting frustrated that it isn't updating and clicking the button multiple times.

Another thought here - we may want to restrict this button so it can only be clicked once per hour or where it checks to see if there is an update process running (not sure if that can be done but I don't see why not). I fear the case where 2 or 3 people click it 2 or 3 times and then we take down the whole site because of all these processes or it breaks events altogether

Reply:
This button runs the task _immediately_ - the page with the success message will not load until after the task has run to completion. This is not ideal behavior as this task takes at least 20 seconds to run, even if no events need to be created - the comment is misleading here. Please let me know if we have queue workers set up on our production instances, which I could then use to run the task in the background. Also, only the EC and higher should have permission to run this so I don't think a time limit is a concern, especially if it is run in a background queue where it wouldn't affect the main site.

Reply:
This button has been entirely removed, marking resolved